cos-fleetshard-operator/src/main/java/org/bf2/cos/fleetshard/operator/connector/ConnectorConfiguration.java

```diff
@@ -0,0 +1,96 @@
+package org.bf2.cos.fleetshard.operator.connector;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.Properties;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.fabric8.kubernetes.client.utils.Serialization;
+
+public class ConnectorConfiguration<S> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConnectorConfiguration.class);
+
+    private final static String PROPERTY_DATA_SHAPE = "data_shape";
+    private final static String PROPERTY_PROCESSORS = "processors";
+    private final static String PROPERTY_ERROR_HANDLER = "error_handler";
+    private static final Set<String> RESERVED_PROPERTIES = Set.of(PROPERTY_DATA_SHAPE, PROPERTY_PROCESSORS,
+        PROPERTY_ERROR_HANDLER);
+    private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
+
+    private final S connectorSpec;
+    private final ObjectNode dataShapeSpec;
+    private final ObjectNode errorHandlerSpec;
+    private final ArrayNode processorsSpec;
+
+    @SuppressWarnings("unchecked")
+    public ConnectorConfiguration(ObjectNode connectorSpec, Class<S> type) {
+        if (null == connectorSpec || connectorSpec.isEmpty()) {
+            throw new RuntimeException("Connector spec can't be empty!");
+        }
+
+        var dataShape = connectorSpec.at("/" + PROPERTY_DATA_SHAPE);
+        if (dataShape.isMissingNode()) {
+            // @TODO Eventually, once most/all connectors support `data_shape` this should become a RuntimeException!
+            LOGGER.warn("Missing `data_shape` in connector spec: " + connectorSpec.toPrettyString());
```

Comment:
@rk3rn3r this may leaks sensitive data

Reply:
@lburgazzoli right. But what else can be logged here to make sense to the user? What do you suggest instead to make context for the user? `connector_id` or other things aren't available here...
My suggestion: throwing an exception again (but a custom one) and catching it in `AbstractOperandController` where we have more context and can add connector name/id at least?

Reply:
if we throw an exception, then the reify process would fail so I guess we are in the same situation as before.
Given that is a temporary solution, maybe we can do the validation/check when the `ConnectorConfiguration` is created ?

Reply:
Do you want me to implement it ?

Reply:
@lburgazzoli  I prepared the Exception, but for now I only log the missing field (without a relation to a specific connector). I think that's okay for now until we can enable it when all connectors support `data_shape` (and maybe make `error_handler` mandatory). I think it should not be too hard to relate it to a connector and it's only temporarily.